examples/energy-management-app/energy-management-common/src/EVSEManufacturerImpl.cpp

```diff
@@ -104,6 +103,223 @@ CHIP_ERROR EVSEManufacturer::Shutdown()
+                    {
+                        addedEnergy_mWh.SetNonNull(chargingTarget.addedEnergy.Value());
+                    }
+                    else
+                    {
+                        addedEnergy_mWh.SetNull();
+                    }
+                }
+            }
+        }
+
+        if (bFound)
+        {
+            // Skip the rest of the search
+            break;
+        }
+    }
+
+    if (!bFound)
+    {
+        err = CHIP_ERROR_NOT_FOUND;
+    }
+
+    return err;
+}
+
+/**
+ * @brief   Simple example to demonstrate how an EVSE can compute the start time
+ *          and duration of a charging schedule
+ */
+CHIP_ERROR EVSEManufacturer::ComputeChargingSchedule()
+{
+    CHIP_ERROR err;
+    EVSEManufacturer * mn = GetEvseManufacturer();
+    VerifyOrReturnError(mn != nullptr, CHIP_ERROR_UNINITIALIZED);
+
+    EnergyEvseDelegate * dg = mn->GetEvseDelegate();
+    VerifyOrReturnError(dg != nullptr, CHIP_ERROR_UNINITIALIZED);
+
+    uint8_t dayOfWeekMap = 0;
+    ReturnErrorOnFailure(GetDayOfWeekNow(dayOfWeekMap));
+
+    uint16_t minutesPastMidnightNow_m = 0;
+    ReturnErrorOnFailure(GetMinutesPastMidnight(minutesPastMidnightNow_m));
+
+    uint32_t now_epoch_s = 0;
+    ReturnErrorOnFailure(GetEpochTS(now_epoch_s));
+
+    DataModel::Nullable<uint32_t> startTime_epoch_s;
+    DataModel::Nullable<uint32_t> targetTime_epoch_s;
+    DataModel::Nullable<Percent> targetSoC;
+    DataModel::Nullable<int64_t> addedEnergy_mWh;
+
+    uint32_t power_W;
+    uint32_t chargingDuration_s;
+    uint32_t tempTargetTime_epoch_s;
+    uint32_t tempStartTime_epoch_s;
+    uint16_t targetTimeMinutesPastMidnight_m;
+
+    // Initialise the values to Null - if the FindNextTarget finds one, then it will update the value
+    targetTime_epoch_s.SetNull();
+    targetSoC.SetNull();
+    addedEnergy_mWh.SetNull();
+    startTime_epoch_s.SetNull(); // If we FindNextTarget this will be computed below and set to a non null value
+
+    /* We can only compute charging schedules if the EV is plugged in and the charging is enabled
+     * so we know the charging current - i.e. can get the max power, and therefore can calculate
+     * the charging duration and hence start time
+     */
+    if (dg->IsEvsePluggedIn() && dg->GetSupplyState() == SupplyStateEnum::kChargingEnabled)
+    {
+        uint8_t searchDay = 0;
+        while (searchDay < 2)
+        {
+            err = FindNextTarget(dayOfWeekMap, minutesPastMidnightNow_m, targetTimeMinutesPastMidnight_m, targetSoC,
+                                 addedEnergy_mWh, (searchDay != 0));
+            if (err == CHIP_ERROR_NOT_FOUND)
+            {
+                // We didn't find one for today, try tomorrow
+                searchDay++;
+                dayOfWeekMap = (dayOfWeekMap << 1) & 0x7f;
+                if (dayOfWeekMap == 0x00)
+                {
+                    dayOfWeekMap = 0x01; // Must be Saturday and shifted off, so set it to Sunday
```

Comment:
need day bitmak constants instead of numeric values.

Reply:
Referenced the chip::app::Clusters::EnergyEvse::TargetDayOfWeekBitmap

Reply:
See above. EnergyEvse::TargetDayOfWeekBitmap is being used instead of a uint8_t